src/main/java/org/wso2/carbon/securevault/azure/handler/AzureKeyVaultSecretCallbackHandler.java

```diff
@@ -0,0 +1,166 @@
+ * Azure Key Vault that stores the deployment secrets.
+ */
+public class AzureKeyVaultSecretCallbackHandler extends AbstractSecretCallbackHandler {
+
+    private static final String ALIAS = "alias";
+    private static final String CONFIG_FILE_PATH = CarbonUtils.getCarbonConfigDirPath() + File.separator +
+            "security" + File.separator + "secret-conf.properties";
+    private static final String DOT = ".";
+    private static final String IDENTITY = "identity";
+    private static final String KEY = "key";
+    private static final String STORE = "store";
+    private static final String PASSWORD = "password";
+    private static final Log log = LogFactory.getLog(AzureKeyVaultSecretCallbackHandler.class);
+    private static String keyStorePassword;
+    private static String privateKeyPassword;
+
+    /**
+     * Handles single secret callback.
+     *
+     * @param singleSecretCallback a single secret callback.
+     */
+    @Override
+    protected void handleSingleSecretCallback(SingleSecretCallback singleSecretCallback) {
+
+        if (keyStorePassword == null && privateKeyPassword == null) {
+            boolean sameKeyAndKeyStorePass = true;
+            String keyPassword = System.getProperty(KEY + DOT + PASSWORD);
+
+            if (keyPassword != null && keyPassword.trim().equals("true")) {
+                sameKeyAndKeyStorePass = false;
+            }
+
+            try {
+                readPasswordFromKeyVault(sameKeyAndKeyStorePass);
+            } catch (AzureKeyVaultException e) {
+                log.error("Building secret client failed: ", e);
+            }
+            readPasswordThroughConsole(sameKeyAndKeyStorePass);
+        }
+
+        if (singleSecretCallback.getId().equals(IDENTITY + DOT + KEY + DOT + PASSWORD)) {
+            singleSecretCallback.setSecret(privateKeyPassword);
+        } else {
+            singleSecretCallback.setSecret(keyStorePassword);
+        }
+    }
+
+    /**
+     * Reads keystore and primary key passwords from Azure Key Vault.
+     *
+     * @param sameKeyAndKeyStorePass flag to indicate whether the keystore and primary key passwords are the same.
+     * @throws AzureKeyVaultException if building a secret client fails.
+     */
+    @SuppressFBWarnings("PATH_TRAVERSAL_IN")
+    private void readPasswordFromKeyVault(boolean sameKeyAndKeyStorePass) throws AzureKeyVaultException {
+
+        if (log.isDebugEnabled()) {
+            log.debug("Reading Carbon Secure Vault configuration properties from file.");
+        }
+
+        InputStream inputStream = null;
+        Properties properties = new Properties();
+
+        try {
+            inputStream = new FileInputStream(CONFIG_FILE_PATH);
+            properties.load(inputStream);
+        } catch (IOException e) {
+            throw new AzureKeyVaultException("Error while loading configurations from " + CONFIG_FILE_PATH + ": ", e);
```

Comment:
Is it valid to throw `AzureKeyVaultException` when it is an IOException while reading file?

Reply:
Fixed.